compiler/tests/data/decimal-singleton.bal

```diff
@@ -0,0 +1,18 @@
+// Zero<:DECIMAL
+// One<:DECIMAL
+// Minus<:DECIMAL
+// ZO<:DECIMAL
+// ZOM<:DECIMAL
+// ZO<:ZOM
+// Zero<:ZO
+// One<:ZO
+// Zero<:ZOM
+// One<:ZOM
+// Minus<:ZOM
+
+type Zero 0d;
```

Comment:
We need some more tests. I suggest some separate files. In particular:

* 0d is the same as -0d
* 1.0d is the same as 1d